Correlation_Module/correlation_sampler.cpp

```diff
@@ -67,7 +67,9 @@ torch::Tensor correlation_sample_forward(
   if (input1.device().is_cuda()){
     CHECK_INPUT(input1);
     CHECK_INPUT(input2);
-    
+    const at::cuda::OptionalCUDAGuard guard_input1(device_of(input1));
```

Comment:
Is it necessary to have a guard for every input ? As far as I understand, the guard is just a way to indicate that every further operation will be on the given device (here, the one `input1` is stored in). The second guard would then overwrite the first one.

I feel like the best solution would be to construct a guard for the `input1` and then check that `input1`and `input2`have the same device, in a new function that could be name e.g. `CHECK_SAME_DEVICE`

Reply:
I see. That's the correct way to do it I think.
I have changed the code to reflect this. Can you have a look?